src/m365/spo/commands/file/file-sharinglink-remove.spec.ts

```diff
@@ -0,0 +1,228 @@
+      }
+    });
+
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified sharing link to a file when confirm option not passed and prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'delete');
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: false }
+    ));
+
+    await command.action(logger, {
+      options: {
+        webUrl: webUrl,
+        fileUrl: fileUrl,
+        id: id
+      }
+    });
+
+    assert(postSpy.notCalled);
+  });
+
+  it('removes specified sharing link to a file by fileId when prompt confirmed', async () => {
+    let sharingLinkRemoveCallIssued = false;
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${webUrl}/_api/web/GetFileById('${fileId}')?$select=SiteId,VroomItemId,VroomDriveId`) {
+        return fileInformationResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/${fileInformationResponse.SiteId}/drives/${fileInformationResponse.VroomDriveID}/items/${fileInformationResponse.VroomItemID}/permissions/${id}`) {
+        sharingLinkRemoveCallIssued = true;
+        return { statusCode: 204 };
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        verbose: true,
+        webUrl: webUrl,
+        fileId: fileId,
+        id: id
+      }
+    });
+    assert(sharingLinkRemoveCallIssued);
+  });
+
+  it('removes specified sharing link to a file by URL', async () => {
+    const fileServerRelativeUrl: string = urlUtil.getServerRelativePath(webUrl, fileUrl);
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${webUrl}/_api/web/GetFileByServerRelativePath(decodedUrl='${formatting.encodeQueryParameter(fileServerRelativeUrl)}')?$select=SiteId,VroomItemId,VroomDriveId`) {
+        return fileInformationResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/${fileInformationResponse.SiteId}/drives/${fileInformationResponse.VroomDriveID}/items/${fileInformationResponse.VroomItemID}/permissions/${id}`) {
+        return { statusCode: 204 };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        verbose: true,
+        webUrl: webUrl,
+        fileUrl: fileUrl,
+        id: id,
+        confirm: true
+      }
+    });
+    assert(loggerLogToStderrSpy.called);
```

Comment:
In fact now we are only testing that we log a `verbose` message.
Let's assert that our deleteStub is called just like I suggested in the previous test.